pygsti/forwardsims/forwardsim.py

```diff
@@ -52,14 +53,23 @@ def cast(cls, obj, num_qubits=None):
 
         if isinstance(obj, ForwardSimulator):
             return obj
-        elif obj == "auto":
-            return _MapFSim() if (num_qubits is None or num_qubits > 2) else _MatrixFSim()
-        elif obj == "map":
-            return _MapFSim()
-        elif obj == "matrix":
-            return _MatrixFSim()
+        elif isinstance(obj, str):
+            if obj == "auto":
+                return _MapFSim() if (num_qubits is None or num_qubits > 2) else _MatrixFSim()
+            elif obj == "map":
+                return _MapFSim()
+            elif obj == "matrix":
+                return _MatrixFSim()
+            else:
+                raise ValueError(f'Unrecognized string argument, {obj}')
+        elif isinstance(obj, Callable):
```

Comment:
Yeah. It's there if someone wants to pass something akin to a factory. Just passing the class name (e.g., ``obj = MatrixForwardSimulator``) is one example of how this might be used. 

This is useful when the forward simulator class of interest is stateful, and hence cannot be shared across different models.